### PR TITLE
[CMD] DATE: Output 'The current date is: '

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose

Make `DATE` command output compatible.
JIRA issue: [CORE-13848](https://jira.reactos.org/browse/CORE-13848)

## Proposed changes

- Add `STRING_DATE_NOW` resource string for localizing `"The current date is: "`.
- Output by `ConOutResPuts` function.

## TODO

- [x] Do tests.

## Comparison

Win2k3:
win2k3

ReactOS (AFTER):
after

### CODE REVIEW

File: base/shell/cmd/resource.h [hunk 1]
```diff
@@ -213,6 +213,7 @@
 #define STRING_RMDIR_HELP2                 724
 #define STRING_MD_ERROR                    725
 #define STRING_MD_ERROR2                   726
+#define STRING_DATE_CURRENT_DATE           727
```
Comment (HBelusca) [inline base/shell/cmd/resource.h:216]: Please rename it as `STRING_DATE_NOW` to follow a similar name as the existing [`STRING_TIME_NOW`](https://git.reactos.org/?p=reactos.git&a=search&h=HEAD&st=grep&s=STRING_TIME_NOW)

File: base/shell/cmd/date.c [hunk 2]
```diff
@@ -203,9 +203,11 @@ INT cmd_date(LPTSTR param)
             nDateString = i;
     }
 
-    // TODO: Should prepend a "Current date is: " prompt, as done with TIME.
     if (nDateString == -1)
+    {
+        ConOutResPrintf(STRING_DATE_CURRENT_DATE); /* "The current date is: " */
```
Comment (HBelusca) [inline base/shell/cmd/date.c:208]:
```suggestion
        ConOutResPuts(STRING_DATE_CURRENT_DATE);
```

File: base/shell/cmd/lang/de-DE.rc [hunk 3]
```diff
@@ -471,6 +471,7 @@ Titel         neuer Titel"
     STRING_GOTO_ERROR2 "Sprungmarke '%s' wurde nicht gefunden\n"
     STRING_MD_ERROR "Unterverzeichnis oder Datei %s existiert bereits.\n"
     STRING_MD_ERROR2 "Der Pfad zum neuen Ordner existiert nicht.\n"//FIXME
+    STRING_DATE_CURRENT_DATE "The current date is: "
```
Comment (JoachimHenze) [inline base/shell/cmd/lang/de-DE.rc:474]:
```suggestion
    STRING_DATE_CURRENT_DATE "Aktuelles Datum: "
```

File: base/shell/cmd/resource.h (hunk 1 above)
Comment (katahiromz) [reply base/shell/cmd/resource.h]: Fixed.

**Review by JoachimHenze — APPROVED** [review]

**Pull request merged**